File: util/cvs2sql.pl
#!/usr/bin/perl

# Example sql to export as CSV
#
#
# SELECT rankings . * 
#FROM tblUserRankings rankings, tblClubUser clubuser
#WHERE rankings.userid = clubuser.userid
#AND rankings.courttypeid =3
#AND rankings.usertype =0
#AND clubuser.clubid =17
#
# usage: perl cvs2sql.pl ~/Downloads/tblUserRankings.sql > output.sql


$start = 6162;
$password = "33f2ee11f1130c5c6e11061cc91d9b9a";
$clubid = 49;
$roleid = 1;
$siteid = 91;


open (MYFILE, $ARGV[0]);
while (<MYFILE>) {
	chomp;
	#last, first, gender, rank
	 
	@personal = split(/,/, $_);
	$firstName = @personal[1];
	chomp($firstName);
	$lastName = @personal[0];
	$email = "";
	$ranking = @personal[3];
	$userName = $firstName.".".$lastName;
	
	
	$gender = @personal[2] eq 'M' ? 0 : 1;

	#users
#	print "INSERT INTO `tblUsers` (`userid`, `username`, `firstname`, `lastname`, `email`, `password`, `gender`, `homephone`)  VALUES ($start,'$userName','$firstName','$lastName','','$password', $gender, '');\n";
	
	#clubuser
	#print "INSERT INTO `tblClubUser` (`userid`, `clubid`, `roleid`, `recemail`)  VALUES ($start,'$clubid','$roleid','n');\n";
		
	#ranking
	print "INSERT INTO `tblUserRankings` (`userid`, `courttypeid`, `ranking`)  VALUES ($start,2,'$ranking');\n";

	#auth
	#print "INSERT INTO `tblkupSiteAuth` (`userid`, `siteid`)  VALUES ($start,$siteid);\n\n";
	
	
	
	++$start;
}
close (MYFILE);

File: util/cvs2sql.py
#!/usr/bin/python

#
# usage:  ./cvs2sql.py ~/Downloads/tblUserRankings.sql > output.sql
# #first, last, email, gender, phone
#
# to rollback
# delete from tblUsers where userid > 8143; delete from tblClubUser where userid > 8143;delete from tblUserRankings where usertype = 0 and userid > 8143;delete from tblkupSiteAuth where userid > 8143

import fileinput
import sys
import csv

# Set these variables for the club
index = 8143
clubid = 55
siteid = 98
ranking = 3
courttype = 7
courttype2 = 9
password = '8f7e08f6ea74964506a8cb2c6cf461eb' #(ladue2014)


#### You probably don't need to do anything below this line ##

filename = sys.argv[-1]
with open(filename, 'rb') as f:
    reader = csv.reader(f)
    for row in reader:

    	firstname = row[0]
    	lastname = row[1].replace("'", "\\'")
    	email = row[2]
    	gender = 1 if row[3] =='F' else '0'
    	username = firstname.lower()+'.'+lastname.lower()
    	phone = row[4]
    	memberid = row[5]

    	#users
    	sys.stdout.write(
    		"INSERT INTO `tblUsers` (`userid`, `username`, `firstname`, `lastname`, `email`, `password`, `gender`, `homephone`,`workphone`,`cellphone`, `pager`, `useraddress`)  VALUES ("+str(index)+",'"+username+"','"+firstname+"','"+lastname+"','"+email+"','"+password+"', '"+str(gender)+"', '"+phone+"','','', '','');\n"
    		)

        #clubuser
    	sys.stdout.write( 
    		"INSERT INTO `tblClubUser` (`userid`, `clubid`, `roleid`, `recemail`, `memberid`)  VALUES ("+str(index)+","+str(clubid)+",'1','n', '"+str(memberid)+"');\n"
    		)

    	# rankings 
    	sys.stdout.write( 
    		"INSERT INTO `tblUserRankings` (`userid`, `courttypeid`, `ranking`)  VALUES ("+str(index)+","+str(courttype)+",'"+str(ranking)+"');\n"
    		)
    	sys.stdout.write( 
			"INSERT INTO `tblUserRankings` (`userid`, `courttypeid`, `ranking`)  VALUES ("+str(index)+","+str(courttype2)+",'"+str(ranking)+"');\n"
		)

    	#auth
    	sys.stdout.write(
    		"INSERT INTO `tblkupSiteAuth` (`userid`, `siteid`)  VALUES ("+str(index)+","+str(siteid)+");\n\n"
    		)

        index+=1


